File: Data Structures & Algorithms/binary-search/submission-0.go
func search(nums []int, target int) int {
    var l, r int = 0, len(nums)-1
    var mid int

    for l <= r {
        mid = l + (r-l)/2

        if nums[mid] == target {
            return mid
        } else if nums[mid] < target {
            l = mid + 1
        } else {
            r = mid-1
        }
    }
    return -1
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        res = []
        intermediate = []
        summation = 0
        length = len(nums)

        def helper(start):
            nonlocal summation

            if summation == target:
                res.append(intermediate[:])
                return
            
            if summation > target:
                return
            
            for index in range(start, length):

                summation += nums[index]
                intermediate.append(nums[index])

                helper(index)

                summation -= nums[index]
                intermediate.pop()

        helper(0)
        return res